client/src/game/units.ts

```diff
@@ -34,7 +34,7 @@ export function l2g(obj: LocalPoint | Ray<LocalPoint>): GlobalPoint | Ray<Global
     const z = gameStore.zoomFactor;
     const panX = gameStore.panX;
     const panY = gameStore.panY;
-    if (obj instanceof LocalPoint) {
+    if (obj instanceof LocalPoint || obj instanceof Point) {
```

Comment:
I had to add this, because the functions .add and .substract from LocalPoint (and GlobalPoint as well, but that does not matter) actually return a Point, which was not being picked up here.

Reply:
looking at the types for LocalPoint and GlobalPoint this should not be true.  They explicitly override the Point implementation for that specific reason as I want to make sure that at no point the two are wrongly used.  Also subtract does not return a Point at all, but a vector

Reply:
Sorry, its only add that was giving me problems.

I know it should be a proper LocalPoint. However, the instanceof was not entering. To see the problem, just checkout the latest commit from this PR and remove this extra condition, and try to copy any shape. It will call the `screenCenter` getter from `store.ts`, which will run `l2g(g2l(this.screenTopLeft).add(halfScreen))`, which will error with the message `Uncaught TypeError: Cannot read property 'origin' of undefined`, which means that `obj instanceof LocalPoint` returned false.

I believe the casting to LocalPoint is only being done at compiler level, but when the actual js runs, a simple Point is returned.

Reply:
I can confirm that the LocalPoint info is lost in js after the add call, which is sad :(.  I'm currently looking into ways to fix this.  Maybe as a last resort I literally reimplement the actual code for both Global and Local points, it's not a lot of code anyway and I would really prefer to not having to add the instanceof Point here as I feel it will lead to unforeseen issues in the future.

Reply:
Ok, got it working! I'm gonna make a quick PR to change the Point implementation and then you should be able to remove the instanceof cast